### PR TITLE
Feature: Ticketmaster Single ID and Email/First Name support.

### PR DESCRIPTION
**Pull request by orospakr**

### CODE REVIEW

File: Sources/Ticketmaster/TicketmasterAuthorizer.swift [hunk 1]
```diff
@@ -8,28 +8,35 @@
...
+    
     /**
      Set the user's Ticketmaster credentials after a successful sign-in with the [Presence SDK](https://developer.ticketmaster.com/products-and-docs/sdks/presence-sdk/). Implement the `onMemberUpdated(backendName:member:)` method in your `PresenceLoginDelegate` and call this method passing in values from the `PresenceMember`.
      
      - Parameters:
-        - accountManagerMemberID: The value of the `PresenceMember`'s `AccountManagerMemberID` property.
-        - hostMemberID: The value of the `PresenceMember`'s `HostMemberID` property.
+     - id: The value of the `PresenceMember`'s `id` property.
+     - firstName: The value of the `PresenceMember`'s `email` property (optional).
+     - firstName: The value of the `PresenceMember`'s `firstName` property (optional).
```
Comment (samwoodall) [inline Sources/Ticketmaster/TicketmasterAuthorizer.swift]: [Nitpick] firstName here twice

File: Sources/Ticketmaster/TicketmasterManager.swift [hunk 2]
```diff
@@ -13,22 +13,77 @@ class TicketmasterManager {
...
+        } else {
+            // neither value was given, treat it as a logout.
+            os_log("Neither accountManagerMemberID or hostMemberID was given to Rover ticketmaster module legacy API.", log: .general, type: .fault)
+            clearCredentials()
+        }
+    }
+    
+    func setCredentials(id: String, email: String?, firstName: String?) {
+        let newMember = Member(id: id, email: email, firstName: firstName)
+        self.member.value = newMember
```
Comment (samwoodall) [inline Sources/Ticketmaster/TicketmasterManager.swift]: You not checking for empty `id` here?

File: Sources/Ticketmaster/TicketmasterManager.swift [hunk 3]
```diff
@@ -13,22 +13,77 @@ class TicketmasterManager {
...
+        
+        // As a side-effect, set the fields into the `ticketmaster` hash in userInfo so they are immediately available even without a server sync succeeding.
+        self.userInfoManager.updateUserInfo {
+            if let existingTicketmasterUserInfo = $0.rawValue["ticketmaster"] as? Attributes {
+                // ticketmaster already exists, just clobber the three fields:
```
Comment (samwoodall) [inline Sources/Ticketmaster/TicketmasterManager.swift]: clobber three fields? or two?

File: Sources/Ticketmaster/TicketmasterManager.swift [hunk 4]
```diff
@@ -95,8 +148,11 @@ extension TicketmasterManager: SyncParticipant {
             return .noData
         }
         
+        let localAttributes: [String: Any] = member.value?.userInfo ?? [String: Any]()
+        
+        // Set the `ticketmaster` field on userInfo, but clobber the id, email, and firstName fields that might have come back from the server with our local values.
```
Comment (samwoodall) [inline Sources/Ticketmaster/TicketmasterManager.swift]: id won't be getting clobbered now?

File: Sources/Ticketmaster/TicketmasterManager.swift (hunk 2 above)
Comment (orospakr) [reply Sources/Ticketmaster/TicketmasterManager.swift]: here’s no optionality concept there so far as I know (empty string or otherwise).

**Comment by orospakr** [comment]
Resolves #48.

**Pull request merged**